From e609a7531ff90685fdfb3f882cc243b9fad8af9e Mon Sep 17 00:00:00 2001
From: Joseph Richard <Joseph.Richard@windriver.com>
Date: Thu, 7 Sep 2017 14:55:44 -0400
Subject: [PATCH 055/155] DB: US102723: Remove unmanaged subnets feature from
 DB

This commit makes the necessary migration to remove unmanaged from
the subnet table.

Conflicts:
	neutron/db/migration/alembic_migrations/versions/EXPAND_HEAD
---
 .../alembic_migrations/versions/EXPAND_HEAD        |  2 +-
 .../wrs_43a0f920c515_remove_managed_from_subnet.py | 43 ++++++++++++++++++++++
 .../alembic_migrations/vswitch_init_ops.py         |  4 ++
 3 files changed, 48 insertions(+), 1 deletion(-)
 create mode 100644 neutron/db/migration/alembic_migrations/versions/pike/expand/wrs_43a0f920c515_remove_managed_from_subnet.py

diff --git a/neutron/db/migration/alembic_migrations/versions/EXPAND_HEAD b/neutron/db/migration/alembic_migrations/versions/EXPAND_HEAD
index 4f00477..75fd3ae 100644
--- a/neutron/db/migration/alembic_migrations/versions/EXPAND_HEAD
+++ b/neutron/db/migration/alembic_migrations/versions/EXPAND_HEAD
@@ -1 +1 @@
-wrs_11a84c4cea76
+wrs_43a0f920c515
diff --git a/neutron/db/migration/alembic_migrations/versions/pike/expand/wrs_43a0f920c515_remove_managed_from_subnet.py b/neutron/db/migration/alembic_migrations/versions/pike/expand/wrs_43a0f920c515_remove_managed_from_subnet.py
new file mode 100644
index 0000000..d8a7dac
--- /dev/null
+++ b/neutron/db/migration/alembic_migrations/versions/pike/expand/wrs_43a0f920c515_remove_managed_from_subnet.py
@@ -0,0 +1,43 @@
+# Copyright 2017 OpenStack Foundation
+#
+#    Licensed under the Apache License, Version 2.0 (the "License"); you may
+#    not use this file except in compliance with the License. You may obtain
+#    a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+#    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+#    License for the specific language governing permissions and limitations
+#    under the License.
+#
+# Copyright (c) 2013-2014 Wind River Systems, Inc.
+#
+# The right to copy, distribute, modify, or otherwise make use
+# of this software may be licensed only pursuant to the terms
+# of an applicable Wind River license agreement.
+#
+
+"""Remove managed from subnet
+
+Revision ID: wrs_43a0f920c515
+Revises: wrs_11a84c4cea76
+Create Date: 2017-09-07 18:54:18.641545
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = 'wrs_43a0f920c515'
+down_revision = 'wrs_11a84c4cea76'
+
+from alembic import op
+
+
+def upgrade():
+    cmd = ("DELETE FROM subnets WHERE id IN"
+           " (SELECT s.id FROM subnets AS s LEFT OUTER JOIN ipallocations"
+           " AS ia ON ia.subnet_id = s.id WHERE s.managed = false"
+           " GROUP BY s.id,ia.ip_address HAVING COUNT(ia.ip_address) = 0);")
+    op.execute(cmd)
+    op.drop_column('subnets', 'managed')
diff --git a/neutron/db/migration/alembic_migrations/vswitch_init_ops.py b/neutron/db/migration/alembic_migrations/vswitch_init_ops.py
index 34cb0dd..89199fe 100644
--- a/neutron/db/migration/alembic_migrations/vswitch_init_ops.py
+++ b/neutron/db/migration/alembic_migrations/vswitch_init_ops.py
@@ -162,3 +162,7 @@ def upgrade():
         sa.ForeignKeyConstraint(['subnet_id'], ['subnets.id'],
                                 ondelete='CASCADE'),
         sa.PrimaryKeyConstraint('id'))
+
+    op.add_column(
+        'subnets',
+        sa.Column('managed', sa.Boolean(), nullable=True))
-- 
2.7.4

